typings/index.d.ts

```diff
@@ -3083,6 +3085,20 @@ declare module 'discord.js' {
     cache?: boolean;
   }
 
+  interface HTTPAttachmentData {
+    name: string;
+    file: Buffer;
```

Comment:
This can also be a stream:
```suggestion
    file: Buffer | Stream;
```
The source for that is a bit hidden:
https://github.com/discordjs/discord.js/blob/99e2ca408b0ead894c2d91bc2902c392a3635fb0/src/util/DataResolver.js#L106-L111